src/chigame/games/models.py

```diff
@@ -367,6 +371,24 @@ def set_archive(self, archive):
         self.archived = archive
         self.save()
 
+    def check_and_end_tournament(self):
+        """
+        Checks if the tournament end date has reached and ends the tournament if it has.
+        Any matches that have not finished are forfeited.
+        """
+        if self.status == "tournament ended":
+            brackets = self.matches.all()
+            for bracket in brackets:
+                assert isinstance(bracket, Match)
+                bracket_users = bracket.players.all()
+                bracket_players = [Player.objects.get(user=user, match=bracket) for user in bracket_users]
+                bracket_with_outcome = any(player.outcome is not None for player in bracket_players)
+                if not bracket_with_outcome:  # the match has not finished
```

Comment:
Hi Elizabeth, the matches field is implemented as such so that the behavior of the tournament detail view will not be too complicated. If all the matches that have been played in the tournament still need to be displayed, then it would be hard to keep track of the rounds in which the matches are played (currently the create_bracket and next_round methods remove all the matches played in the last round and create new matches for the next round). Therefore, I think that the matches field should be used to show all matches in that round instead of showing the status of some matches. 

Still, it is a great point that we should consider how to handle forfeited matches. An individual deadline for each round may be added later so that matches are forfeited in each round and do not impact the process of moving to the next round.

Reply:
Hey Mike, that logic makes sense to me. However, my point is mostly that the phrasing "no matches" is not intuitive and appears to be saying that this was a tournament without any matches at all (whether they were played or not). I think it should be changed to at least something like "matches forfeited" to indicate that there were in fact matches, but they were not played.